mlflow/deployments/cli.py

```diff
@@ -12,29 +12,34 @@ def _user_args_to_dict(user_list):
         except ValueError:
             # not enough values to unpack
             raise click.BadOptionUsage("config", "Config options must be a pair and should be"
-                                                 "provided as ``--C key=value`` or "
+                                                 "provided as ``-C key=value`` or "
                                                  "``--config key=value``")
         if name in user_dict:
             raise click.ClickException("Repeated parameter: '{}'".format(name))
         user_dict[name] = value
     return user_dict
 
 
-target_uri = click.option("--target-uri", "t_uri", required=True,
-                          help="Deployment target URI. Check the documentation/help for each "
-                               "plugin to understand the target URI format the plugins expect. "
-                               "For fetching the help, you can call "
-                               "`mlflow deployments help --target-name <target-name>`")
-target_name = click.option("--target-name", "t_name", required=True,
-                          help="Deployment target name.")
+registered = "\n".join(["* %s" % target for target in interface.plugin_store.registry])
+
+target_details = click.option("--target", "-t", required=True,
+                              help="Deployment target URI. Check the documentation/help for each "
+                                   "plugin to understand the target URI format the plugins "
+                                   "expect. Run "
+                                   "`mlflow deployments help --target-name <target-name>` for "
+                                   "more details on the supported URI format and config options "
+                                   "for a given target."
+                                   "Support is currently installed for the following targets (see"
+                                   " other deployment targets and installation instructions in "
+                                   "https://mlflow.org/docs/latest/plugins.html#community-plugins"
+                                   "): {}".format(registered))
```

Comment:
The formatting of the helpstring for the `--target` argument currently looks a bit odd (looks like `click` ignores the newline characters), maybe we could make it more readable via something like: 

```suggestion
registered = [target for target in interface.plugin_store.registry]

target_details = click.option("--target", "-t", required=True,
                              help="""
                                   Deployment target URI. Check the documentation/help for each "
                                   plugin to understand the target URI format the plugins 
                                   expect. Run 
                                   `mlflow deployments help --target-name <target-name>` for 
                                   more details on the supported URI format and config options 
                                   for a given target. 
                                   Support is currently installed for the following targets: 
                                   {}. 
                                   
                                   See other deployment targets and installation instructions in 
                                   https://mlflow.org/docs/latest/plugins.html#community-plugins
                                   """.format(registered))
```